datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -160,11 +160,47 @@ impl<'a> TypeCoercionRewriter<'a> {
         op: Operator,
         right: Expr,
     ) -> Result<(Expr, Expr)> {
+        // try to coerce the scalar to the column type
+        fn coerce_scalar_to_col(
```

Comment:
I think the previous logic that calls `get_input_types` should handle all types of Exprs, including scalars.

Thus I am not sure we need special case logic here for just scalars (which won't work for other argument types)

Did you look into extending `get_input_types` to handle Utf8Views?

Reply:
Yes, I did look at that. But the problem is that `get_input_types` only handles `DataType` instead of `Expr`

```rust
/// Returns the coerced input types for a binary expression evaluating the `op` with the left and right hand types
pub fn get_input_types(
    lhs: &DataType,
    op: &Operator,
    rhs: &DataType,
) -> Result<(DataType, DataType)> {
    signature(lhs, op, rhs).map(|sig| (sig.lhs, sig.rhs))
}
```

Reply:
I think @Weijun-H is right that we need to consider not only coercion by data type but also their column types.
Normally: 
`(Utf8View, Utf8) => (Utf8View, Utf8View)`

But:
`(Scalar(Utf8View), Column(Utf8)) => (Scalar(Utf8), Column(Utf8)) `

I'll take a closer look and think about what is the best way to approach this..

Reply:
🤔  I wonder if this might be a good candidate for extending the UnwrapCastInComparison logic, which already handles similar cases with scalars 

https://github.com/apache/datafusion/blob/ed7c884d64b2750529ec6b256cdbab582341de07/datafusion/optimizer/src/unwrap_cast_in_comparison.rs#L38-L74

Reply:
It could be a workaround, I will look into it.